src/blob_storage.h

```diff
@@ -115,6 +129,23 @@ class BlobStorage {
 
   // Only BlobStorage OWNS BlobFileMeta
   std::unordered_map<uint64_t, std::shared_ptr<BlobFileMeta>> files_;
+
+  class InternalComparator {
+   public:
+    InternalComparator() = default;
+    InternalComparator(const Comparator* comparator)
```

Comment:
remove the default constructor and add `explicit` keyword for the constructor with one param.